packages/tax/src/services/tax-module-service.ts

```diff
@@ -196,6 +196,27 @@ export default class TaxModuleService<
     return await this.taxRateService_.update({ selector, data }, sharedContext)
   }
 
+  async upsert(
+    data: TaxTypes.UpsertTaxRateDTO[],
+    sharedContext?: Context
+  ): Promise<TaxTypes.TaxRateDTO[]>
+  async upsert(
+    data: TaxTypes.UpsertTaxRateDTO,
+    sharedContext?: Context
+  ): Promise<TaxTypes.TaxRateDTO>
+
+  @InjectTransactionManager("baseRepository_")
+  async upsert(
+    data: TaxTypes.UpsertTaxRateDTO | TaxTypes.UpsertTaxRateDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TaxTypes.TaxRateDTO | TaxTypes.TaxRateDTO[]> {
+    const result = await this.taxRateService_.upsert(data, sharedContext)
```

Comment:
This will vary from module to module, but many times we'd like to do some validation and normalization to the input before doing update/create, so calling the internal create_ and update_ methods might be better. It might be fine here though, just wanted to mention it